src/Projekt/Args.fs

```diff
@@ -1,111 +1,90 @@
-                | Some org -> org
-                | None -> "MyOrg"
-            Init (ProjectInitData.create (path, template, fxv, org)) |> Success
-            
-        | ToLower "addfile" :: FullPath project :: FullPath file :: Options opts ->
-            let compile = opts.GetResult(<@ Compile @>, true)
-            AddFile { ProjPath = project
-                      FilePath = file
-                      Link = opts.TryGetResult <@ Link @>
-                      Compile = compile }
-            |> Success
-            
-        | [ToLower "delfile"; FullPath project; FullPath file] -> 
-            DelFile { ProjPath = project; FilePath = file }
-            |> Success
-            
-        | ToLower "movefile" :: FullPath project :: FullPath file :: Options opts
-                when opts.Contains <@ Direction @> ->
+    [<Verb("delfile", HelpText = "Delete a file from a project")>]
+    type public DelFileOptions = 
+        {   [<Value(0, Required = true, MetaName = "project path")>]    projectPath : string
+            [<Value(1, Required = true, MetaName = "file path")>]       filePath : string   }
+    with
+        member x.ToOperation =
+            match x.projectPath, x.filePath with
+            | FullPath project, FullPath file -> 
+                // typing needed here because of the duplication between MoveFileData and DelFileData records
+                // TODO: maybe consolidate?
+                ({ProjPath = project; FilePath = file} : DelFileData) |> DelFile 
+            | _,_ -> failwith "invalid paths"
 
-            let direction = opts.PostProcessResult(<@ Direction @>, parseDirection)
-            MoveFile { ProjPath = project
-                       FilePath = file
-                       Direction = direction
-                       Repeat = opts.GetResult(<@ Repeat @>, 1)}
-            |> Success
-            
-        | [ToLower "reference"; FullPath project; FullPath reference] -> 
-            Reference { ProjPath = project; Reference = reference } |> Success
+    let parse args = 
+        let parsed = CommandLine.Parser.Default.ParseArguments<InitOptions, ReferenceOptions, MoveFileOptions, AddFileOptions, DelFileOptions>(args)
+        // tried to get fancy here with a statically resolved type param to invoke the ToOperation member on the individal option cases, but I couldn't get it to work....
 
-        | _ -> Failure (parser.Usage (sprintf "Error: '%s' is not a recognized command or received incorrect arguments.\n\n%s" args.Head commandUsage))
-    with
-    | :? System.ArgumentException as e ->
-            let lines = e.Message.Split([|'\n'|])
-            let msg = parser.Usage (sprintf "%s\n\n%s" lines.[0] commandUsage)
-            Failure msg
+        parsed.Return<InitOptions, ReferenceOptions, MoveFileOptions, AddFileOptions, DelFileOptions, Result<Operation>>(
+            (fun (init : InitOptions) -> init.ToOperation |> Success), 
```

Comment:
I think they are, since this call is a C#-style method call which needs the whole batch of tupled arguments .  If I remove the parens and the separating commas the call doesn't compile.